### DISCUSSION TITLE
Can I use root as a package location with workspaces?

### DISCUSSION THREAD

**Discussion by jansiegel** [npm/feedback#96] [CLI] [answered]
I'm not sure if it's a bug or lack of feature, so I'm posting it here.

Is it safe to declare `root` as a package location with npm workspaces? For example:
```
"workspaces": {
  "packages": [
    ".",
    "packages/*"
  ]
}
```

I can't find anything about doing so in the RFCs or the documentation (if I missed something, point me in the right direction), so I figured it should work fine - and it does for the most part, the dependencies seem to install alright, they get linked properly, but I'm having an issue with generating `package-lock.json`.
Namely, When I declare `"."` as a package location, `package-lock` gets created, but it looks more like a slighly-modified `package.json`, than a regular lock file:
```
{
  "name": "workspaces-test",
  "version": "1.0.0",
  "lockfileVersion": 2,
  "requires": true,
  "packages": {
    "": {
      "version": "1.0.0",
      "license": "ISC",
      "devDependencies": {
        "rimraf": "^2.5.4"
      },
      "workspaces": {
        "packages": [
          ".",
          "packages/*"
        ]
      }
    }
  }
}
```

Is this a bug, or should I use the root exclusively as a top-level config location?

**Comment by arcanis** [comment]
In Yarn the manifest that defines a `workspaces` field is necessarily a workspace itself, so I'd expect you not to have to explicitly define `.` in your array

**Reply by ljharb** [reply]
In lerna, you definitely do have to define `.` - i use that in one lerna project, and do not use it in another; it seems important that both be possible.

**Comment by MylesBorins** [comment, marked as answer]
@jansiegel would you be able to open this question as an issue on https://github.com/npm/cli

**Reply by jansiegel** [reply]
Sure, will do
